src/index.ts

```diff
@@ -496,7 +497,7 @@ export class Loader {
       key: this.apiKey,
       channel: this.channel,
       client: this.client,
-      libraries: this.libraries,
+      libraries: this.libraries.length && this.libraries,
```

Comment:
what about:

```
libraries: this.libraries.length > 0 ? this.libraries : null
```

Delete l 508:

```
    Object.keys(params).forEach(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (key) => !(params as any)[key] && delete (params as any)[key]
    );
```

Update l 541
```
// from
for (k in g) e.set(k.replace(/[A-Z]/g, (t) => "_" + t[0].toLowerCase()), g[k]);
// to
for (k in g) {
  if (g[k] != null) {
    e.set(k.replace(/[A-Z]/g, (t) => "_" + t[0].toLowerCase()), g[k]);
  }
}

Reply:
Do we actually need to pass the libs in there as they will be imported via `importLibrary` anyway ?

Reply:
[sorry, looks like email-based replies don't thread nicely]

> Doesn't that make my point even more valid?
> 
> You are more likely to have a cache hit if the url doesn't specify any library (as it will be common whatever set of lib you use).
> 
> That's possible because the loader `importLibrary` of all individual library anyway.
> 
> (Though it probably depends on how `google.maps.importLibrary` is implemented)

For performance, importLibrary (when used with the inline bootstrap) can inject libraries into the underlying bootstrap URL to be loaded (it very briefly collects an initial batch to be sent, which causes them to load faster). However, if you have two different pages on your site, one which first uses `importLibrary('maps')` and the other which first uses `importLibrary('places')` then the bootstrap URLs will end up being different. To super-optimize caching, one might decide to set `libraries: ['maps', 'places']` across all pages (at the tradeoff of loading more data up front). Or one might set `libraries: ['core']` across all pages (at the tradeoff of needing extra follow-up requests in serial for maps or places). That's just one example, there are other situations where fine-tuning could be desirable. 

For most sites, this is probably a premature optimization, but the option remains for those who want to fine-tune the performance. Generally I would recommend configuring with no libraries, and then calling the `importLibrary()` calls one needs as they need them.